Polyfills/Canvas/Source/Canvas.cpp

```diff
@@ -5,11 +5,17 @@
 #include <functional>
 #include <sstream>
 #include <assert.h>
+#include <NativeEngine.h>
 
 namespace Babylon::Polyfills::Internal
 {
     static constexpr auto JS_CONSTRUCTOR_NAME = "NativeCanvas";
 
+    struct NativeCanvas::Impl final
```

Comment:
I don't understand this. Why do you need an Impl class?

Reply:
I added an impl to have a data container so that i wouldn't have to declare the babylon/NativeEngine.h header in the the canvas header.

Reply:
tking another look it seems like there are two canvas headers and we already declare bgfx types in the internal canvas header. I'll remove this container and will just declare the texturedata as a property of the canvas struct for now.

Reply:
this should now be fixed. impl is removed and we just declare the property for the native canvas.

Reply:
based on other conversations, this datas lifecycle is going to be maintained by the external through a finalizer.

Reply:
that would make a lot more sense. create it for javascript and finalize with javascript.

Reply:
this should now be fixed